pkg/release/hook.go

```diff
@@ -41,13 +43,22 @@ type HookDeletePolicy string
 
 // Hook delete policy types
 const (
-	HookSucceeded          HookDeletePolicy = "succeeded"
-	HookFailed             HookDeletePolicy = "failed"
+	HookSucceeded          HookDeletePolicy = "hook-succeeded"
```

Comment:
would you mind explaining the reason behind this change? It looks like this may break backwards compatibility with charts that defined their hook deletion policies, which encompasses way more than `helm test`

Reply:
This is actually to ensure backwards compatibility. 😄 

Confusingly, `pkg/hooks` had its own definitions for deletion policies, which are refactored out in this PR:

https://github.com/helm/helm/blob/539bafa58b6ecb68f5a387a6fda0708d31723ab5/pkg/hooks/hooks.go#L46-L51